tests/core/test_selector.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+from pytest_mock.plugin import MockerFixture
+
+from sqlmesh.core import dialect as d
+from sqlmesh.core.environment import Environment
+from sqlmesh.core.model import Model, SqlModel
+from sqlmesh.core.selector import ModelSelector
+from sqlmesh.core.snapshot import SnapshotChangeCategory
+from sqlmesh.utils import UniqueKeyDict
+
+
+def test_model_selection(mocker: MockerFixture, make_snapshot):
```

Comment:
Could you also write some tests covering behavior when models have dependencies on each other? I know we ended up with this low-level approach to where that doesn't really matter but it would be good to assert that expected behavior.